src/api/app/controllers/staging/staged_requests_controller.rb

```diff
@@ -1,5 +1,81 @@
-class Staging::StagedRequestsController < Staging::ProjectsController
+class Staging::StagedRequestsController < ApplicationController
+  before_action :require_login, except: [:index]
+  before_action :set_staging_project
+  before_action :set_staging_workflow, :set_project, :set_xml_hash, only: [:create, :destroy]
+
   def index
-    @requests = @project.staged_requests
+    @requests = @staging_project.staged_requests
+  end
+
+  def create
+    authorize @staging_project, :update?
+    requests = @staging_workflow.unassigned_requests.where(number: request_numbers)
+
+    result = ::Staging::StageRequests.new(requests: requests, staging_project_name: @staging_project.name).perform
+    @staging_project.staged_requests << result
+    unassigned_requests = request_numbers - result.pluck(:number).map(&:to_s)
+
+    if unassigned_requests.empty?
+      render_ok
+    else
+      render_error(
+        status: 400,
+        errorcode: 'invalid_request',
+        message: "Could not assign requests #{unassigned_requests.join(', ')} to #{@staging_project}."
+      )
+    end
+  end
+
+  def destroy
+    authorize @staging_project, :update?
+    requests = @staging_project.staged_requests.where(number: request_numbers)
+    package_names = requests.joins(:bs_request_actions).pluck('bs_request_actions.target_package')
+
+    @staging_project.staged_requests.delete(requests)
```

Comment:
Is there any reason why we want to skip the callbacks here? :thinking:

Reply:
Yes, we only want to remove the association, so destroy does not work here because it would destroy the requests objects.

Reply:
I think you mixed the 

https://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-delete

with

https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-delete